packages/bot-web-ui/src/stores/route-prompt-dialog-store.js

```diff
@@ -0,0 +1,38 @@
+import { observable, action } from 'mobx';
+
+export default class RoutePromptDialogStore {
+    constructor(root_store) {
+        this.root_store = root_store;
+    }
+
+    @observable should_show = false;
+    @observable is_confirmed = false;
+    @observable last_location = null;
+
+    @action.bound
+    handleBlockedNavigation(next_location) {
```

Comment:
Since it returns a boolean, could we name it as such? e.g. `shouldNavigateAfterPrompt`

Reply:
sure